### PR TITLE
Minor housekeeping related to deprecated methods

### PR DESCRIPTION
**Pull request by musm**
Might be  cleaner to put these in the deprecated.jl file? If you think it's more logical to keep it as is feel free to close!

### CODE REVIEW

**Comment by Ward9250** [comment]
LGTM! :+1:

File: src/deprecated.jl [hunk 1]
```diff
@@ -2,3 +2,24 @@ typealias EmptyStreamSource EmptyStream
...
+
+if VERSION < v"0.6-" 
+    function Base.takebuf_array(stream::BufferedOutputStream{EmptyStream})
+        chunk = stream.buffer[1:stream.position-1]
+        stream.position = 1
+        return chunk
+    end
+end
+
+if v"0.5-" < VERSION < v"0.6-"
```
Comment (tkelman) [inline src/deprecated.jl]: the lower bound isn't specific enough here, early 0.5 prereleases will not recognize String as a concrete type

Comment (Ward9250) [inline src/deprecated.jl]: I guess that as simple a change as `v"0.5-"` -> `v"0.5"` should be sufficient to fix this?

Comment (tkelman) [inline src/deprecated.jl]: that should be good enough

Comment (musm) [inline src/deprecated.jl]: fixed by https://github.com/BioJulia/BufferedStreams.jl/pull/31

Comment (Ward9250) [inline src/deprecated.jl]: Thanks, merged and a patch release made!

**Pull request merged**